src/ast/passes/codegen_llvm.cpp

```diff
@@ -531,6 +532,28 @@ ScopedExpr CodegenLLVM::visit(NegativeInteger &integer)
   return ScopedExpr(b_.getInt64(integer.value));
 }
 
+ScopedExpr CodegenLLVM::visit(NamedParameter &named_param)
```

Comment:
I'm not a huge fan of needing new custom codegen for this case. Why can't this be handled as standard map after desugaring? (And in codegen maybe we add some map metadata that says whether soemthing is read-only, and we reject writes to "read-only" maps in the semantic analyzer.)

Reply:
Yeah, that's not a bad idea. Let me look into that.

Reply:
Done.